### PR TITLE
Adding igbp vegetation type to model and visualization

### PR DESCRIPTION
**Pull request by jsignell**
Added land cover to the visualization options and to the model
veg_type_colored_by_site

### CODE REVIEW

**Comment by jbednar** [comment]
Thanks!  Looks cool.  Can you please add that content to the original notebook and make the PR against that notebook, not a separate file like this?  The notebooks should also be cleared; you can put one on anaconda.org to show us what the output should be.

**Comment by jsignell** [comment]
I cleared all the output from the notebook in this PR and renamed the notebook based on the pre-existing file. The changes reflect work done by @stsievert as well. I wasn't quite sure how to properly handle a notebook in git. 

I put the executed version of the notebook here: https://anaconda.org/jsignell/carbon_flux/notebook

File: examples/Carbon_Flux.ipynb [hunk 1]
```diff
@@ -36,43 +42,179 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DAILIES_DIR = '../../nee_data_fusion/data/in_situ/fluxnet_daily/'\n",
+    "METADATA_CSV = '../../nee_data_fusion/data/in_situ/extracted/allflux_metadata.txt'\n",
+    "# DAILIES_DIR = 'flux_data/dailies/'\n",
+    "# METADATA_CSV = 'allflux_metadata.txt'\n",
```
[the remark this answers is not in the document]
Comment (jbednar) [reply examples/Carbon_Flux.ipynb]: We generally put our datasets on S3 and then we have a data catalog that declares which ones we need.  This is fine for now; we'll update it to our standard format later.

File: examples/Carbon_Flux.ipynb [hunk 2]
```diff
@@ -81,65 +223,411 @@
...
+    "\n",
+    "# add metadata to the big dataframe\n",
+    "df = pd.merge(df, metadata, on='site')\n",
+    "df = pd.merge(df, categorical_igbp_mapper, on='igbp')\n",
+    "\n",
+    "print(\"df shape =\", df.values.shape)\n",
+    "if float('nan') in df.columns:\n",
+    "    print('nan in df.columns, removing')\n",
+    "    df.pop(float('nan'))"
+   ]
```
Comment (jbednar) [inline examples/Carbon_Flux.ipynb]: What does a column `nan` mean?

File: examples/Carbon_Flux.ipynb [hunk 3]
```diff
@@ -81,65 +223,411 @@
...
-    "* NEE_CUT_USTAR50 : Net Ecosystem Exchange, using Constant Ustar Threshold (CUT) across years, from 50 percentile of USTAR threshold\n",
-    "* NEE_VUT_USTAR50: Net Ecosystem Exchange, using Variable Ustar Threshold (VUT) for each year, from 50 percentile of USTAR threshold\n",
+    "We want to generate some visualization that accounts for these 4 variables and helps generate some understanding.\n",
     "\n",
-    "Other variables should not be included as features, either because they are derived from NEE:\n",
+    "That is, these observations lie on some manifold. We want to learn the structure of that manifold, and visualize each observation on that manifold.\n",
     "\n",
-    "* GPP_DT_VUT_USTAR50: Gross Primary Production, from Daytime partitioning method, based on NEE_VUT_USTAR50\n",
-    "* GPP_DT_CUT_USTAR50: Gross Primary Production, from Daytime partitioning method, based on NEE_CUT_USTAR50\n",
+    "This will have the finding similar observations that have a similar structure between the indepedent variables (e.g., `P_ERA`) and depedent variables (the carbon flux measurement `y`).\n",
```
Comment (jbednar) [inline examples/Carbon_Flux.ipynb]: Can't parse this sentence. Also: indepedent -> independent, depedent -> dependent

File: examples/Carbon_Flux.ipynb [hunk 4]
```diff
@@ -81,65 +223,411 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "igbp_vegetation = {\n",
+    "    'ENF': '01 - Evergreen Needleleaf forest',\n",
+    "    'EBF': '02 - Evergreen Broadleaf forest',\n",
+    "    'DNF': '03 - Deciduous Needleleaf forest',\n",
+    "    'DBF': '04 - Deciduous Broadleaf forest',\n",
+    "    'MF': '05 - Mixed forest',\n",
```
Comment (jbednar) [inline examples/Carbon_Flux.ipynb]: Maybe add a space after : to line it up.

File: examples/Carbon_Flux.ipynb [hunk 5]
```diff
@@ -81,65 +223,411 @@
...
     "\n",
-    "Here, we will focus on predicting ``NEE_CUT_USTAR50``. \n",
     "\n",
-    "Hypothesis : RSIF is more about vegetation than solar flux. Idea is that RSIF reflects carbon capture by vegetation.\n",
+    "The rest of this notebook is copy-pasted from `Carbon-FLux.ipynb`. I haven't modified it yet.\n",
     "\n",
-    "NEE: Soil flux and photosynthesis of carbon.\n",
-    "Note: Not trying to separate these contributions.\n",
-    "Goal: Finding features. Adding features."
+    "`----------------------------------------------------`"
```
Comment (jbednar) [inline examples/Carbon_Flux.ipynb]: These two lines can be omitted, right?  It's now been integrated?

File: examples/Carbon_Flux.ipynb [hunk 6]
```diff
@@ -81,65 +223,411 @@
...
+    "        show.columns\n",
+    "    ))\n",
+    "))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "These variables are sufficient to create the linear models at every site. However, the site information is hidden from the visualization algoritm.\n",
```
Comment (jbednar) [inline examples/Carbon_Flux.ipynb]: algoritm -> algorithm.  Also (below):

preform -> perform
inlude -> include
lattitude -> latitude
ecentricities -> eccentricities
tranform -> transform
wrwapping -> wrapping

File: examples/Carbon_Flux.ipynb (hunk 2 above)
Comment (jsignell) [reply examples/Carbon_Flux.ipynb]: @stsievert I am taking this part out unless you know why it is there. I agree that it doesn't really make sense.

File: examples/Carbon_Flux.ipynb (hunk 3 above)
Comment (jsignell) [reply examples/Carbon_Flux.ipynb]: @stsievert do you have any changes for this?

File: examples/Carbon_Flux.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbednar) [reply examples/Carbon_Flux.ipynb]: Sounds good to me.  That can happen after this PR.

**Pull request merged**